modular_zubbers/code/modules/security/security_glock/gun.dm

```diff
@@ -1,28 +1,125 @@
+	. += span_notice("Spin reloads can be done on the move and need you only to hold your gun for their 2 second duration. ")
+	. += span_notice("If toggled on, spin reloads throw empty magazines in the direction you're facing, dealing damage dependant on their type. ")
+
+/obj/item/gun/ballistic/automatic/pistol/sec_glock/attack_self(mob/living/user)
+	var/obj/item/storage/our_belt = user.get_item_by_slot(ITEM_SLOT_BELT)
+	var/obj/item/ammo_box/magazine/m9mm/security/swapped_mag = locate(/obj/item/ammo_box/magazine/m9mm/security) in our_belt.contents
+	if(!magazine && swapped_mag) //Empty gun, load in anyhow but without delay. No mag
+		if(insert_magazine(user, swapped_mag))
+			SpinAnimation(2, 1)
+			balloon_alert(user, "you spin your gun, loading in a mag!")
+		else
+			balloon_alert(user, "that magazine doesn't fit!")
+	if(!chambered)
+		if(swapped_mag)
+			balloon_alert(user, "you spin your gun, loading in a mag!")
+			SpinAnimation(2, 1)
+			if(do_after(user, 1 SECONDS, timed_action_flags = (IGNORE_USER_LOC_CHANGE | IGNORE_TARGET_LOC_CHANGE), target = src))
+				throw_eject_magazine(user, magejecting, swapped_mag)
+		else
+			balloon_alert(user, "no spare magazines in your belt!")
+			return
+	drop_bolt(user)
+	if(recent_rack > world.time)
+		return
+	recent_rack = world.time + rack_delay
+	rack(user)
+
+
+/obj/item/gun/ballistic/automatic/pistol/sec_glock/proc/throw_eject_magazine(mob/user, display_message = TRUE, obj/item/ammo_box/magazine/m9mm/security/belt_mag, atom/target)
+	if(bolt_type == BOLT_TYPE_OPEN)
+		chambered = null
+	if(magazine.ammo_count())
+		playsound(src, eject_sound, eject_sound_volume, eject_sound_vary)
+	else
+		playsound(src, eject_empty_sound, eject_sound_volume, eject_sound_vary)
+	magazine.forceMove(drop_location())
+	var/obj/item/ammo_box/magazine/m9mm/security/old_mag = magazine
+	if(belt_mag)
+		insert_magazine(user, belt_mag)
+	else
+		magazine = null
+		return
+	old_mag.forceMove(drop_location())
+	playsound(old_mag, old_mag.murphy_eject_sound, 50, TRUE)
+	if(magejecting)
+		old_mag.was_ejected = 1
+		old_mag.throw_at(get_edge_target_turf(user, user.dir), range = 7, speed = 7, thrower = user, force = 10)
+	old_mag.update_appearance()
+	update_appearance()
+
+/obj/item/gun/ballistic/automatic/pistol/sec_glock/insert_magazine(mob/user, obj/item/ammo_box/magazine/AM, display_message = TRUE)
+	if(!istype(AM, accepted_magazine_type))
+		balloon_alert(user, "[AM.name] doesn't fit!")
+		return FALSE
+	user.transferItemToLoc(AM, src)
+	magazine = AM
+	if(bolt_type == BOLT_TYPE_OPEN && !bolt_locked)
+		chamber_round(TRUE)
+	drop_bolt()
+	update_appearance()
+	..()
+	return
 
-/obj/item/gun/ballistic/automatic/pistol/sec_glock/security/rubber //This is what you give to cargo packages.
-	spawn_magazine_type = /obj/item/ammo_box/magazine/m9mm/rubber
+// for your viewing convenience, same procs, just without bubbles
+/obj/item/gun/ballistic/automatic/pistol/sec_glock/drop_bolt(mob/user = null)
```

Comment:
btw I meant to override the original drop_bolt without bubble messages, even if it is not modular, it's worth doing here, via arg on drop_bolt and rack